internal/domain/users/usersStore_test.go

```
@@ -144,6 +144,35 @@ var _ = Describe("UsersStore", func() {
 		})
 
 		// TODO: add tests for GetUserByResetToken
+		Describe("GetUserByResetToken", func() {
+			BeforeEach(func() {
+				_, err = db.Exec(`update users
+								  set reset_token = 'reset token'
+								  where id = $1`, expectedUserInfo.ID)
+				Expect(err).To(BeNil())
+			})
+
+			DescribeTable("GetUserByResetToken", func(
+				inputResetToken string,
+				expectHasRecord bool,
+				expectedHasError bool,
+			) {
+				actualUser, actualHasRecord, err := usersStore.GetUserByResetToken(context.Background(), inputResetToken)
+				if expectedHasError {
+					Expect(err).To(HaveOccurred())
+				} else {
+					Expect(err).To(BeNil())
+					Expect(actualHasRecord).To(Equal(expectHasRecord))
+					if expectHasRecord {
+						Expect(actualUser.ID).To(Equal(expectedUserInfo.ID))
+						Expect(actualUser.ResetToken.String).To(Equal("reset token"))
+					}
+				}
+			},
+				Entry("should successfully get user info with valid reset token", "reset token", true, false),
+				Entry("should return false, no err when the reset token doesn't exist in db", "invalid reset token", false, false),
```

Can we also have tests for when an error should be thrown? 
Like an empty reset token should return an error, I believe.

---

Yes. Thank you for pointing it out! 
Added